src/pysdmx/model/__base.py

```diff
@@ -27,6 +29,24 @@ class Annotation(Struct, frozen=True, omit_defaults=True):
     url: Optional[str] = None
     type: Optional[str] = None
 
+    def __post_init__(self) -> None:
+        """Additional validation checks for Annotation."""
+        if (
+            not self.id
```

Comment:
Maybe some general method is needed to avoid this combination, I will try to look for sth but is not blocking.

Reply:
Thanks, @javihern98. My understanding is that you will search for another way to avoid this combination but, in the mean time, the proposed check is good enough, and so I will mark this as resolved. Happy to discuss if I misunderstood :-)